pkgs/tools/misc/pre-commit/default.nix

```diff
@@ -70,6 +97,14 @@ buildPythonApplication rec {
       --subst-var-by virtualenv ${virtualenv}
     substituteInPlace pre_commit/languages/node.py \
       --subst-var-by nodeenv ${nodeenv}
+    substituteInPlace pre_commit/clientlib.py \
+      --subst-var-by nodeenv ${nodeenv}
+
+    # meta-hook-rewiring needs a git repo, not just a source dir
+    # let it have one in the nix store
+    echo SRC ${src}
+    substituteInPlace pre_commit/clientlib.py \
```

Comment:
This substitution is supposed to populate the REPO_ROOT variable, but as far as I can tell it doesn't work either

Reply:
Superficially it looks like you're missing surrounding @'s in the patch. See https://nixos.org/manual/nixpkgs/stable/#fun-substitute.